### DISCUSSION TITLE
Cannot generate metrics for vedantmgoyal2009

### DISCUSSION THREAD

**Discussion by vedantmgoyal9** [lowlighter/metrics#950] [Q&A] [answered]
Hi there,

I have observed that https://metrics.lecoq.io/ cannot generate metrics for me but it can generate for other people.

Why is it so? 😔

Thanks.

**Comment by vedantmgoyal9** [comment]
@lowlighter I have also tried using a VPN, but still, I'm getting errors.

**Comment by lowlighter** [comment]
Hi 👋 !

From what I've seen, some of your stats are too high and they result in API timeouts (this can be seen with `NaN` stats below, and the total contributed repositories does not even display the number)

<img src="https://user-images.githubusercontent.com/22963968/159403549-acc7b44b-5d3e-4dd1-bbbf-bab62c859c17.png" width="400">

Your account seems to be able to be generated through the GitHub action (result above) and maybe it'll be possible to fill out gaps for while fixing #951. So you could eventually use it instead.

However I think unfortunately you won't be able to generate these using directly the website. The reason is that to workaround API timeouts, it is required to perform multiples queries instead of a single one, which consume a lots of API requests and resource time, which is why the website actually timeout too

Hope it helps 👍

**Reply by vedantmgoyal9** [reply]
Actually I was using the website to preview how would my selected configuration would look, and then copy GitHub Adtion code from there.

**Reply by lowlighter** [reply]
Even without generating the live preview, you should be able to click on the `Action code` tab and create a new workflow with code displayed.

I think it's the best you can achieve currently 😅

**Reply by vedantmgoyal9** [reply]
Can you please convert this config to a markdown template and output it directly to README.md at the root of the repository. Actually, I couldn't understand the parameters with the markdown template. Classic template configuration is simple but it's difficult to understand how the markdown template works.

![metrics](https://user-images.githubusercontent.com/83997633/160174911-68ebdbc8-59c4-4f5d-9e79-e1e0612462f8.jpg)

**Reply by lowlighter** [reply]
The website doesn't provide the templating to use the markdown template anyway, but the documentation for it is located [here](https://github.com/lowlighter/metrics/blob/master/source/templates/markdown/README.md).

Basically you need to create a `TEMPLATE.md` first, and you can then use [EJS syntax](https://ejs.co) to render variables and sub-graphs.

For example:
```ejs
<!-- TEMPLATE.md -->

Use fetched data:
<%= plugins.introduction.text %>
<%= plugins.traffic.views.count %>

Embed SVG image
<%- await embed(`unique-id`, {habits:true, habits_charts:true} %>

Include one of markdown supported plugins
<%- await include(`partials/topics.ejs`) %>
```

```yml
# workflow.yml
uses: lowlighter/metrics@latest
with:
  template: markdown
  filename: README.md 
  markdown: https://raw.githubusercontent.com/user/repo/branch/TEMPLATE.md
  token: ${{ secrets.TOKEN }}
  config_output: markdown
  plugin_habits: yes
  plugin_introduction: yes
  plugin_traffic: yes
  plugin_topics: yes
```

This [example.md](https://github.com/lowlighter/metrics/blob/master/source/templates/markdown/example.md) can give you an idea of what it is supposed to look like

**Comment by vedantmgoyal9** [comment]
I would rather want to know if there is a setting through which I can directly output SVG to README.md

**Reply by lowlighter** [reply]
You can just use the link of the rendered SVG in your README.md directly like below:
```md
![Metrics](/github-metrics.svg)
```

Since it's a link, each time the SVG is updated people will always load the newest render

**Comment by vedantmgoyal9** [comment, marked as answer]
Is there any way to change the path of SVG? I don't want it in the root of the repository.

**Reply by lowlighter** [reply]
See [`filename`](https://github.com/lowlighter/metrics/blob/master/source/plugins/core/README.md#filename) option for this 🙂

**Reply by vedantmgoyal9** [reply]
Oh, it's output path. I thought it was just to rename github-metics.svg to <something-else>.svg. Thanks!🎉
